File: repos_estudiantes_TARQUI_primera_entrega/MALLCU_CALANI_ALEJANDRO_GABRIEL/POLIMORFISMO/Ejercicio 7/Animal.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.animal;

/**
 *
 * @author USUARIO
 */
class Animal {
    public String hacerSonido() {
        return "Este animal hace un sonido.";
    }

    public String moverse() {
        return "Este animal se mueve de alguna manera.";
    }
}

class Perro extends Animal {
    @Override
    public String hacerSonido() {
        return "Guau guau";
    }

    @Override
    public String moverse() {
        return "Corre";
    }
}

class Gato extends Animal {
    @Override
    public String hacerSonido() {
        return "Miau miau";
    }

    @Override
    public String moverse() {
        return "Salta";
    }
}

class Pajaro extends Animal {
    @Override
    public String hacerSonido() {
        return "Pío pío";
    }

    @Override
    public String moverse() {
        return "Vuela";
    }
}

public class Main {
    public static void main(String[] args) {
        Animal perro = new Perro();
        Animal gato = new Gato();
        Animal pajaro = new Pajaro();

        Animal[] animales = {perro, gato, pajaro};
        for (Animal animal : animales) {
            System.out.println("Sonido: " + animal.hacerSonido());
            System.out.println("Movimiento: " + animal.moverse());
        }
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/POLIIMORFISMO/EJERCICIO 7/Animal.py
class Animal:
    def hacer_sonido(self):
        print("Sonido genérico de animal.")

    def moverse(self):
        print("Movimiento genérico de animal.")

class Perro(Animal):
    def hacer_sonido(self):
        print("El perro dice: ¡Guau guau!")

    def moverse(self):
        print("El perro corre alegremente.")

class Gato(Animal):
    def hacer_sonido(self):
        print("El gato dice: ¡Miau!")

    def moverse(self):
        print("El gato salta y camina con elegancia.")

class Pajaro(Animal):
    def hacer_sonido(self):
        print("El pájaro dice: ¡Pío pío!")

    def moverse(self):
        print("El pájaro vuela por el cielo.")

def main():
    perro = Perro()
    gato = Gato()
    pajaro = Pajaro()

    animales = [perro, gato, pajaro]

    for animal in animales:
        animal.hacer_sonido()
        animal.moverse()
        print("-" * 40)

if __name__ == "__main__":
    main()
